### ISSUE TITLE
Path traversal vulerability using query params as Map instead of direct resolution

### ISSUE THREAD

**Issue by preetkaran20** [SasanLabs/VulnerableApp#301]
**Describe the bug**
Currently path traversal vulnerability is using query params with Map: https://github.com/SasanLabs/VulnerableApp/blob/master/src/main/java/org/sasanlabs/service/vulnerability/pathTraversal/PathTraversalVulnerability.java#L93-L94 which should be changed to springboot queryparam injection directly.

In this issue, please also look at the overall vulnerability, check if there are things like hints, attack vectors, vulnerbaility descriptions are missing and correct them

**Comment by shreelakshmijoshi** [comment]
Hi @preetkaran20 !
I am a new open source contributor and would love to solve this issue, could you help me understand the issue better by providing some extra references ? 
Is this is issue open to contribute ?

**Comment by preetkaran20** [comment]
Hi @shreelakshmijoshi,

Thanks for showing interest in taking this issue.
The issue is that https://github.com/SasanLabs/VulnerableApp/blob/master/src/main/java/org/sasanlabs/service/vulnerability/pathTraversal/PathTraversalVulnerability.java#L93-L94 class is using `Map<String,String>`  for query params, however, we can directly inject using ```@RequestParam String fileName```. 
If this issue is there with any other class too, we need to fix them too.

Thanks,
Karan

**Comment by shreelakshmijoshi** [comment]
In PathTraversalVulnerability.java  , 
should Map<String, String> be directly passed as a parameter  instead of @RequestParam?
Before :
> public ResponseEntity<GenericVulnerabilityResponseBean<String>> getVulnerablePayloadLevel1(
```           @RequestParam Map<String, String> queryParams) { ```
        String fileName = queryParams.get(URL_PARAM_KEY);
        return this.readFile(() -> fileName != null, fileName);
    }

After :
> public ResponseEntity<GenericVulnerabilityResponseBean<String>> getVulnerablePayloadLevel1(
  ```          Map<String, String> queryParams) {  ```
        String fileName = queryParams.get(URL_PARAM_KEY);
        return this.readFile(() -> fileName != null, fileName);
    }

I might be sounding very stupid..

**Comment by preetkaran20** [comment]
it should be 
```java
public ResponseEntity<GenericVulnerabilityResponseBean> getVulnerablePayloadLevel1(
@RequestParam Map<String, String> queryParams)
```

to

```java
public ResponseEntity<GenericVulnerabilityResponseBean> getVulnerablePayloadLevel1(
@RequestParam String fileName)
```

actual point is that getting all the parameters in map then then extracting is not required and we can directly inject the value.

thanks,
Karan

**Comment by shreelakshmijoshi** [comment]
Oh okay.. got it..
Since 
 ```
  private static final String URL_PARAM_KEY = "fileName";
```
is already defined in the class , we don't have to use Map.
Thanks a lot ! @preetkaran20

**Comment by preetkaran20** [comment]
No, it is because of spring injections. You can compare the two injections one with the above file vs https://github.com/SasanLabs/VulnerableApp/blob/master/src/main/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjection.java#L91 one.
we wanted to do something similar to the later one.
thanks,
Karan
